### DISCUSSION TITLE
How can we use tailwindcss with react-native in application development?

### DISCUSSION THREAD

**Discussion by bensadia-med** [tailwindlabs/tailwindcss#3737] [Help]
Hi there,

I'm developing a mobile application and I'm looking to style it with Tailwindcss, so is there any chance to make it without relying on other GitHub repositories

Thanks.

**Comment by simonswiss** [comment]
Hey!

The issue here is Tailwind CSS produces a CSS output, and in the world of React Native, [you cannot use CSS directly](https://reactnative.dev/docs/style). Sure, you can use flex layout properties with [Yoga](https://yogalayout.com/), but that's not real CSS, it's an implementation of Flexbox APIs.

There are currently a few libraries that reproduce what Tailwind CSS can do, but with a JavaScript output.

Tailwind CSS in its current form is not capable of that, so you won't be able to use it directly in React Native projects, unfortunately.

It's not entirely out of question that _one day_ a compiler for Tailwind that generates React Native compatible JavaScript styles could be created, but there is definitely no concrete plans at this point to implement such a thing 👍

**Comment by crherman7** [comment]
@bensadia-med there are 2+ react native implementations for Tailwind CSS - https://github.com/jaredh159/tailwind-react-native-classnames & https://github.com/vadimdemedes/tailwind-rn.

@simonswiss thank you and Tailwind Labs for this great lib - w/ continued dev, support and maintenance! If react native is on the road map for first level support it may be worth looking into https://github.com/styled-components/css-to-react-native. None the less, love Tailwind!
